File: DSA_python/DSA03006 SẮP XẾP THAM LAM.py
def submit():
    n = int(input())
    A = list(map(int, input().split()))

    B = sorted(A)

    def check():
        for i in range(n):
            if A[i] != B[i] and A[i] != B[n-i-1]:
                return False
        return True
    
    print("Yes" if check() else "No")

test = int(input())
for i in range(test):
    submit()

File: DSA/DSA03006 SẮP XẾP THAM LAM.cpp
#include <bits/stdc++.h>
using namespace std;

bool check(vector<int> &A, vector<int> &B, int n)
{
    for (int i = 0; i < n; i++)
    {
        if (A[i] != B[i] && A[i] != B[n - i - 1])
        {
            return false;
        }
    }
    return true;
}

void submit()
{
    int n;
    cin >> n;
    vector<int> A(n);
    for (auto &&i : A)
    {
        cin >> i;
    }
    vector<int> B(A.begin(), A.end());
    
    sort(B.begin(), B.end());

    check(A, B, n) ? (cout << "Yes") : (cout << "No");
    

    cout << endl;
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    int test;
    cin >> test;
    while (test--)
    {
        submit();
    }
    

    return 0;
}